symengine/cwrapper.cpp

```diff
@@ -62,6 +62,10 @@ void basic_free_heap(basic_struct *s)
     delete s;
 }
 
+TypeID basic_get_type(const basic s) {
+    return (TypeID) s->m->get_type_code();
```

Comment:
One should use probably `reinterpret_cast` or whatever the proper C++ cast is. One should not use the raw C cast in C++ codes.